tools/explorer/pkg/escrow/resource.go

```diff
@@ -88,11 +89,14 @@ func (e Stellar) processReservationResources(resData workloads.ReservationData)
 	for nodeID, rsu := range rsuPerNodeMap {
 		node, err := e.nodeAPI.Get(e.ctx, e.db, nodeID, false)
 		if err != nil {
-			return nil, errors.Wrap(err, "could not get node")
+			return nil, false, errors.Wrap(err, "could not get node")
+		}
+		if node.FreeToUse {
```

Comment:
I don't think this is correct. Now if any of the workloads is deployed on a free to use node, then the full reservation is marked as free. It should be the opposite.

Reply:
I though we were going to validate the reservation upfront? So all the nodes for a specific reservation here would be freeToUse or not